File: Semestru 3/MAP/lab/lab2/src/decorator/DelayTaskRunner.java
package decorator;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class DelayTaskRunner extends AbstractTaskRunner {
    private static final Integer miliseconds = 3000;


    public DelayTaskRunner(TaskRunner taskRunner) {
        super(taskRunner);
    }

    @Override
    public void executeOneTask() {
        try {
            Thread.sleep(this.miliseconds);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        super.executeOneTask();
    }
}

File: Semestru 3/MAP/lab/lab9/lab9/lab9/decorator/DelayTaskRunner.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using lab9.model;
namespace lab9.decorator
{
    internal class DelayTaskRunner : AbstractTaskRunner
    {
        private static int miliseconds = 3000;

        public DelayTaskRunner(ITaskRunner runner)
            : base(runner)
        {

        }

        public void ExecuteOneTask()
        {
            Thread.Sleep(miliseconds);
            base.ExecuteOneTask();
        }
    }
}
